pkg/fqdn/dnspoller_test.go

```diff
@@ -35,114 +35,108 @@ import (
+			checkFunc: func(lookups map[string]int, gen *RuleGen) {
 				c.Assert(len(lookups), Equals, 2, Commentf("More than two DNS names was looked up for a rule with 2 DNS name"))
 				for _, name := range []string{dns.Fqdn("cilium.io"), dns.Fqdn("github.com")} {
 					c.Assert(lookups[name], Not(Equals), 0, Commentf("No lookups for DNS name %s in rule", name))
 					c.Assert(lookups[name], Equals, 1, Commentf("More than one DNS lookup was triggered for the same DNS name %s", name))
 				}
 			},
-			rulesToAdd:    []*api.Rule{rule3},
-			rulesToDelete: nil,
+			selectorsToAdd:    api.FQDNSelectorSlice{ciliumIOSel, githubSel},
+			selectorsToDelete: nil,
 		},
 
 		{
 			desc:                        "Lookup name once when two rules refer to it",
 			lookupIterationsAfterAdd:    1,
 			lookupIterationsAfterDelete: 0,
-			checkFunc: func(lookups map[string]int, generatedRules []*api.Rule, gen *RuleGen) {
+			checkFunc: func(lookups map[string]int, gen *RuleGen) {
 				c.Assert(len(lookups), Equals, 1, Commentf("More than one DNS name was looked up for a rule with 1 DNS name"))
 				for _, name := range []string{dns.Fqdn("cilium.io")} {
 					c.Assert(lookups[name], Not(Equals), 0, Commentf("No lookups for DNS name %s in rule", name))
 					c.Assert(lookups[name], Equals, 1, Commentf("More than one DNS lookup was triggered for the same DNS name %s", name))
 				}
 			},
-			rulesToAdd:    []*api.Rule{rule1, rule2},
-			rulesToDelete: nil,
+			selectorsToAdd:    api.FQDNSelectorSlice{ciliumIOSel, ciliumIOSel},
+			selectorsToDelete: nil,
 		},
 
 		{
 			desc:                        "No lookups after removing all rules",
 			lookupIterationsAfterAdd:    0,
 			lookupIterationsAfterDelete: 1,
-			checkFunc: func(lookups map[string]int, generatedRules []*api.Rule, gen *RuleGen) {
+			checkFunc: func(lookups map[string]int, gen *RuleGen) {
 				c.Assert(len(lookups), Equals, 0, Commentf("DNS lookups occurred after removing all rules"))
 			},
-			rulesToAdd:    []*api.Rule{rule1},
-			rulesToDelete: []*api.Rule{rule1},
-		},
-
-		{
-			desc:                        "One lookup for a name after removing one of two referring rules",
-			lookupIterationsAfterAdd:    0,
-			lookupIterationsAfterDelete: 1,
-			checkFunc: func(lookups map[string]int, generatedRules []*api.Rule, gen *RuleGen) {
-				c.Assert(len(gen.GetDNSNames()), Equals, 1, Commentf("Incorrect number of DNS targets for single name with a single referring rule"))
-				c.Assert(len(lookups), Equals, 1, Commentf("Incorrect number of lookups for single name with a single referring rule"))
-				for _, name := range []string{dns.Fqdn("cilium.io")} {
-					c.Assert(lookups[name], Not(Equals), 0, Commentf("No lookups for DNS name %s in rule", name))
-					c.Assert(lookups[name], Equals, 1, Commentf("More than one DNS lookup was triggered for the same DNS name %s", name))
-				}
-			},
-			rulesToAdd:    []*api.Rule{rule1, rule2},
-			rulesToDelete: []*api.Rule{rule2},
```

Comment:
I'll add it back - it's not a very interesting case since we track FQDNs by inserting them into a set (e.g., without duplicates), but it's good to ensure that we actually are testing that the set membership is working as expected.